packages/api/src/objectSet/EXPERIMENTAL_ObjectSetListener.ts

```diff
@@ -26,10 +27,9 @@ export interface EXPERIMENTAL_ObjectSetListener<
   onChange?: (objects: Array<OsdkObjectOrInterfaceFrom<O>>) => void;
 
   /**
-   * An objects backing datasource has changed and a new value is provided
+   * A property on an object that is backed by another datasource has received changes
    */
-  // TODO: Implement
-  onReferenceUpdate?: (updates: Array<OsdkObjectOrInterfaceFrom<O>>) => void;
+  onReferenceUpdate?: (updates: Array<ReferenceUpdate<O>>) => void;
```

Comment:
I am wondering why I would want to get an array of things to deal with vs something that looks like a partial object?

Reply:
If I understand what you're saying, Saurav and I talked about this yesterday, since the GTSR property types are going to look like the timeseries properties we already have, we can't really return the actual value as a part of the object where the property is then typed with the actual value instead of the shape of the handlers